salt/states/pkg.py

```diff
@@ -1725,6 +1890,247 @@ def installed(
+            - advisory_ids:
+              - SUSE-SLE-SERVER-12-SP2-2017-185
+              - SUSE-SLE-SERVER-12-SP2-2017-150
+              - SUSE-SLE-SERVER-12-SP2-2017-120
+    '''
+    if isinstance(advisory_ids, list) and len(advisory_ids) == 0:
+        return {'name': name,
+                'changes': {},
+                'result': True,
+                'comment': 'No advisory ids provided'}
+
+    # Only downloading not yet downloaded packages
+    targets = _find_advisory_targets(name, advisory_ids, **kwargs)
+    if isinstance(targets, dict) and 'result' in targets:
+        return targets
+    elif not isinstance(targets, list):
+        return {'name': name,
+                'changes': {},
+                'result': False,
+                'comment': 'An error was encountered while checking targets: '
+                           '{0}'.format(targets)}
+
+    comment = []
+    if __opts__['test']:
+        summary = ', '.join(targets)
+        comment.append('The following advisory patches would be '
+                       'downloaded: {0}'.format(summary))
+        ret = {'name': name,
+               'changes': {},
+               'result': None,
+               'comment': '\n'.join(comment)}
+        return ret
+
+    changes = {}
+    try:
+        pkg_ret = __salt__['pkg.install'](name=name,
+                                          advisory_ids=advisory_ids,
+                                          downloadonly=downloadonly,
+                                          **kwargs)
+        changes.update(pkg_ret)
+    except CommandExecutionError as exc:
+        ret = {'name': name, 'result': False}
+        if exc.info:
+            # Get information for state return from the exception.
+            ret['changes'] = exc.info.get('changes', {})
+            ret['comment'] = exc.strerror_without_changes
+        else:
+            ret['changes'] = {}
+            ret['comment'] = ('An error was encountered while downloading '
+                              'package(s): {0}'.format(exc))
+        return ret
+
+    if not changes and not comment:
+        status = 'downloaded' if downloadonly else 'installed'
+        comment.append('Related packages are already {}'.format(status))
+
+    ret = {'name': name,
+           'changes': changes,
+           'result': True,
+           'comment': '\n'.join(comment)}
+    return ret
+
+
+def patch_downloaded(name, advisory_ids=None, **kwargs):
+    '''
+    Ensure that packages related to certain advisory ids are downloaded.
+
+    Currently supported for the following pkg providers:
+    :mod:`yumpkg <salt.modules.yumpkg>` and :mod:`zypper <salt.modules.zypper>`
+
+    CLI Example:
+
+    .. code-block:: yaml
+
+        preparing-to-fix-issues:
+          pkg.patch_downloaded:
+            - advisory_ids:
+              - SUSE-SLE-SERVER-12-SP2-2017-185
+              - SUSE-SLE-SERVER-12-SP2-2017-150
+              - SUSE-SLE-SERVER-12-SP2-2017-120
+    '''
```

Comment:
Like the others, let's gate this to prevent it from being used on platforms that don't support it.